charts/kubernetes-agent/values.yaml

```diff
@@ -60,16 +45,45 @@ agent:
   # -- The log level of the agent. Logs are written to the pod logs as well as to file
   # @section -- Agent values
   logLevel: "Info"
-  # -- The default Kubernetes namespace for deployments
-  # @section -- Agent values
-  defaultNamespace: ""
   # -- The number of polling TCP connections to open with the target Octopus Server
   # @section -- Agent values
   pollingConnectionCount: 5
-
   # -- True if events should be scraped and added to the metrics config map
   # @section -- Agent values
   enableMetricsCapture: true
+  
+  deploymentTarget:
+    # -- Set to register the agent as a Deployment Target using the provided initial values
+    # @section -- Agent as Deployment Target values
+    enabled: false
+    initial:
+      # -- The deployment target environments to register the agent with
+      # @section -- Agent as Deployment Target initial values
+      environments: [ ]
+      # -- The deployment target roles to register the agent with
+      # @section -- Agent as Deployment Target initial values
+      tags: [ ]
+      # -- Can be `Untenanted`, `TenantedOrUntenanted` or `Tenanted`.
+      # @section -- Agent as Deployment Target initial values
+      tenantedDeploymentParticipation: "Untenanted"
+      # -- The deployment target tenants to register the agent with
+      # @section -- Agent as Deployment Target initial values
+      tenants: [ ]
+      # -- The deployment target tenant tags to register the agent with
+      # @section -- Agent as Deployment Target initial values
+      tenantTags: [ ]
+      # -- The default Kubernetes namespace for deployments
+      # @section -- Agent as Deployment Target initial values
+      defaultNamespace: ""
+    
+  worker:
+    # -- Set to register the agent as a Worker using the provided initial values
+    # @section -- Agent as Worker values
+    enabled: false
+    initial:
+      # -- The worker pool IDs to associate with the worker
+      # @section -- Agent as Worker initial values
+      workerPoolIds: [ ]
```

Comment:
Lets go with` workerPools` I think... (without `Ids`)

Reply:
yes good point. Since we are allowing names ids or slugs